app/src/tests/components/contents/PacketReadPermission/PacketReadPermission.test.tsx

```diff
@@ -0,0 +1,95 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import { rest } from "msw";
+import { MemoryRouter, Route, Routes } from "react-router-dom";
+import { SWRConfig } from "swr";
+import { PacketReadPermission } from "../../../../app/components/contents/PacketReadPermission/PacketReadPermission";
+import { PacketOutlet } from "../../../../app/components/main/PacketOutlet";
+import { UserState } from "../../../../app/components/providers/types/UserTypes";
+import { UserProvider } from "../../../../app/components/providers/UserProvider";
+import { HttpStatus } from "../../../../lib/types/HttpStatus";
+import { manageRolesIndexUri } from "../../../../msw/handlers/manageRolesHandlers";
+import { server } from "../../../../msw/server";
+import { mockNonUsernameRolesWithRelationships, mockPacket, mockUserState } from "../../../mocks";
+
+const mockGetUserFromLocalStorage = jest.fn((): null | UserState => null);
+jest.mock("../../../../lib/localStorageManager", () => ({
+  ...jest.requireActual("../../../../lib/localStorageManager"),
+  getUserFromLocalStorage: () => mockGetUserFromLocalStorage()
+}));
+
+const renderComponent = () =>
+  render(
+    <SWRConfig value={{ dedupingInterval: 0, provider: () => new Map() }}>
+      <UserProvider>
+        <MemoryRouter initialEntries={[`/${mockPacket.name}/${mockPacket.id}/read-permissions`]}>
+          <Routes>
+            <Route element={<PacketOutlet packetId={mockPacket.id} />}>
+              <Route path="/:packetName/:packetId/read-permissions" element={<PacketReadPermission />} />
+            </Route>
+          </Routes>
+        </MemoryRouter>
+      </UserProvider>
+    </SWRConfig>
+  );
+
+describe("PacketReadPermission", () => {
```

Comment:
Could test that it works with both global and specific packet manage permission?

Reply:
Didn't think that was needed as the logic for checking if user can manage packet  is in `canManagePacket`,